test/test-pipe-connect-error.c

```diff
@@ -43,15 +43,16 @@ static void close_cb(uv_handle_t* handle) {
 
 
 static void connect_cb(uv_connect_t* connect_req, int status) {
-  ASSERT(status == UV_ENOENT);
-  uv_close((uv_handle_t*)connect_req->handle, close_cb);
+  ASSERT_EQ(status, UV_ENOENT);
+  uv_close((uv_handle_t*) connect_req->handle, close_cb);
   connect_cb_called++;
 }
 
 
 static void connect_cb_file(uv_connect_t* connect_req, int status) {
-  ASSERT(status == UV_ENOTSOCK || status == UV_ECONNREFUSED);
-  uv_close((uv_handle_t*)connect_req->handle, close_cb);
+  if (status != UV_ENOTSOCK)
+      ASSERT_EQ(status, UV_ECONNREFUSED);
```

Comment:
```suggestion
    ASSERT_EQ(status, UV_ECONNREFUSED);
```

Reply:
I am inclined to just allow `UV_EACCES` as an error in this case too. WDYT? It happens because the file is read-only, which does not affect the error code on *BSD, but does alter the error code on linux.
```suggestion
    if (status != UV_EACCES)
      ASSERT_EQ(status, UV_ECONNREFUSED);
```

Reply:
Good idea.